File: 1-ioc_di/src/main/java/it/italiancoders/demodi/components/ReportExec.java
package it.italiancoders.demodi.components;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class ReportExec {

    @Autowired
    private List<String> studentsList;

    public void execute(){
        System.out.println("I nostri studenti sono");

        for(String student : studentsList){
            System.out.println("Nome:"+student);

        }
    }
}

File: 1-ioc_di-kotlin/src/main/kotlin/it/italiancoders/demodi/components/ReportExec.kt
package it.italiancoders.demodi.components

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class ReportExec @Autowired constructor(private val studentsList: List<String>) {
    fun execute() {
        println("I nostri studenti sono")

        for (student in studentsList) {
            println("Nome:$student")
        }
    }
}